forecast_solar/__init__.py

```diff
@@ -112,6 +112,11 @@ async def _request(
             data = await response.json()
             raise ForecastSolarRatelimit(data["message"])
 
+        if response.status == 502:
```

Comment:
Should we raise an error for this case already before the check if status is less than 500?

Reply:
Moved above the less then 500